cmd/format_partition.go

```diff
@@ -10,18 +12,70 @@ var (
 		Short: "Formats a partition",
 		Long: "Formats a partition with your choice of filesystem",
 		Run: func(cmd *cobra.Command, args []string) {
-			formatPartition()
+			if (GetString(cmd, "device") == "" && GetString(cmd, "filesystem-device") == "") {
+				logger.Fatal("Either --device or --filesystem-device are required.")
+			}
+
+			if (GetString(cmd, "device") != "" && GetUint(cmd, "partition") == 0) {
+				logger.Fatal("When using the --device parameter, the --partition number must be specified.")
+			}
+
+			formatPartition(
+				GetString(cmd, "device"),
+				GetString(cmd, "filesystem-device"),
+				GetUint(cmd, "partition"),
+				GetString(cmd, "format"),
+				GetString(cmd, "mount-point"),
+				GetStringSlice(cmd, "options"),
+			)
 		},
 	}
 )
 
 func init() {
-	formatPartitionCmd.PersistentFlags().String("disk", "/dev/sda", "Disk containing partition")
-	formatPartitionCmd.PersistentFlags().Int("partition", 1, "Partition number to be formatted")
-	formatPartitionCmd.PersistentFlags().String("filesystem", "ext4", "Filesystem to be applied to the partition")
+	formatPartitionCmd.PersistentFlags().String("device", "", "Block device")
+	formatPartitionCmd.PersistentFlags().String("filesystem-device", "", "Filesystem Block device")
```

Comment:
whats the difference between device and filesystem-device? Does device get partitioned and then filesystem-device is one of the partition's block device?

Reply:
I'm a little bit out of the loop still here. How does this tool get used? Can everything be done in one invocation from unpartitioned block dev -> many fs on top of raid? If not then I don't see the reason for device and fs-device. If so it still feels like format doesn't really need to know about both the dev and the fs-dev.